db/config.cc

```diff
@@ -361,6 +361,10 @@ db::config::config(std::shared_ptr<db::extensions> exts)
         "If set to higher than 0, ignore the controller's output and set the compaction shares statically. Do not set this unless you know what you are doing and suspect a problem in the controller. This option will be retired when the controller reaches more maturity")
     , compaction_enforce_min_threshold(this, "compaction_enforce_min_threshold", liveness::LiveUpdate, value_status::Used, false,
         "If set to true, enforce the min_threshold option for compactions strictly. If false (default), Scylla may decide to compact even if below min_threshold")
+    , compaction_flush_all_tables_before_major_seconds(this, "compaction_flush_all_tables_before_major_seconds", value_status::Used, 86400,
+        "Set the minimum interval in seconds between flushing all tables before each major compaction (default is 86400). "
+        "This option is useful for maximizing tombstone garbage collection by releasing all active commitlog segments. "
+        "Set to 0 to disable automatic flushing all tables before major compaction")
```

Comment:
I didn't understand this description. The default is 86400 seconds, i.e., a day. What does it mean? If I do a "nodetool compact" it will or won't flush the tables? What does "a day" do here - does it mean that if the previous nodetool compact happened more than a day or less than a day, the behavior is different?

Reply:
Yes, flush of **all** tables happens at most once a day by default since it is expensive and may create small sstables due to excessive memtable flushing.
The reason is to deal with the user / management tool invoking the `keyspace_compaction` api repeatedly, for each keyspace. It's sufficient to flush all tables only once, on the first invocation.
If more than a day passed since last flush then all tables would be flush again.

BTW, this is related to https://github.com/scylladb/scylladb/pull/15971 that will help flush idle memtable in a timely manner, and with it this mechanism here should skip tables that were already flushed in the last 24 hours, but we're not yet committed to it.